core/src/SYCL/Kokkos_SYCL_Instance.cpp

```diff
@@ -75,9 +75,7 @@ void SYCLInternal::initialize(const sycl::device& d) {
       Kokkos::Impl::throw_runtime_exception(
           "There was an asynchronous SYCL error!\n");
   };
-  // FIXME SYCL+Cuda Apparently, submit_barrier doesn't quite work as expected
-  // for oneAPI 2023.0.0 on NVIDIA GPUs.
-#if defined(KOKKOS_IMPL_ARCH_NVIDIA_GPU)
+#ifdef KOKKOS_IMPL_SYCL_USE_IN_ORDER_QUEUES
   initialize(
```

Comment:
Why did you drop the maco guard and the comment?

Reply:
I moved it to `Kokkos_Setup_SYCL.hpp` since that is the central place where we decide to use in-order or out-of-order queues.